Source/WebKit/Shared/Cocoa/CoreIPCSecureCoding.mm

```diff
@@ -29,9 +29,70 @@
 #import "CoreIPCSecureCoding.h"
 
 #import "ArgumentCodersCocoa.h"
+#import "AuxiliaryProcessCreationParameters.h"
+#import "WKCrashReporter.h"
+#import <WebCore/RuntimeApplicationChecks.h>
+#import <wtf/text/StringHash.h>
 
 namespace WebKit {
 
+namespace SecureCoding {
+
+static bool& internalShouldCrashOnSecureCoding()
+{
+    static dispatch_once_t onceToken;
+    static bool crashOnSecureCoding { false };
+    dispatch_once(&onceToken, ^{
+        if (WebCore::isInAuxiliaryProcess())
+            return;
+
+        crashOnSecureCoding = !![[NSUserDefaults standardUserDefaults] objectForKey:@"WebKitCrashOnSecureCodingKey"];
+    });
+
+    return crashOnSecureCoding;
+}
+
+static HashSet<String>& internalClassNamesExemptFromSecureCodingCrash()
+{
+    static dispatch_once_t onceToken;
+    static NeverDestroyed<HashSet<String>> exemptClassNames;
+    dispatch_once(&onceToken, ^{
+        if (!internalShouldCrashOnSecureCoding())
+            return;
+
+        NSArray *array = [[NSUserDefaults standardUserDefaults] arrayForKey:@"WebKitCrashOnSecureCodingKey"];
+        if (!array)
+            return;
+
+        for (id object in array) {
```

Comment:
May want to make sure it's actually an NSArray before iterating.

Reply:
That's guaranteed by this NSUserDefaults API (if it's not an array then nil is returned.)
But I'll put a release assert in.